pkg/plugin/query.go

```diff
@@ -541,22 +541,23 @@ func iterateRows(buf []byte, readRowFn func(reader *IonReader, index int) error)
 
 		err = reader.StepIn()
 		if err != nil {
-			break
+			return nil, err
 		}
 
 		err = readRowFn(reader, index)
 		if err != nil {
-			break
+			return nil, err
 		}
 
 		err = reader.StepOut()
 		if err != nil {
-			break
+			return nil, err
 		}
-
 		index++
 	}
-
+	if status == nil {
```

Comment:
This is already checked on a higher level (might make sense to check it here instead, but in this case you might want to remove the existing check).

Reply:
The nil check isn't present in `deriveSchema`, which is why there was a panic:
```go
	status, err := iterateRows(buf, func(reader *IonReader, index int) error {
		schema.RowCount += 1
		return analyzeRow(reader, &schema, lookup)
	})
	if err != nil {
		return nil, err
	}
	schema.FinalStatus = status
	// Detect missing values
	for _, col := range schema.Columns {
		if col.Count != schema.RowCount {
			col.Optional = true
		}
	}
	if status.ResultSet.IsEmpty() { // <-- panic if status == nil and err == nil
		return &schema, nil
	}
```

Reply:
Ahh makes sense! The existing check is on an even higher level.